### DISCUSSION TITLE
Multi-line field descriptions printing newline

### DISCUSSION THREAD

**Discussion by tgomesfork** [graphql/graphql-js#4009] [Q&A]
Following [this `type-graphql` discussion](https://github.com/MichalLytek/type-graphql/discussions/1600#discussioncomment-8023277), since `graphql-js` v16.0.0, multi-line field descriptions do not seem to be printed correctly.
```diff
-  """
-  Get the custom reservation list configuration
-  Capability required: `custom_reservation_list`
-  """
+  "\nGet the custom reservation list configuration\nCapability required: `custom_reservation_list`\n"
```
Previously we were able to define them like this:
```typescript
@Query(() => CustomReservationList, {
  description: `
    Get the custom reservation list configuration
    Capability required: \`custom_reservation_list\`
  `,
```
Now to get the proper printed schema, we need to put the `\n` ourselves on the description:
```typescript
  @Query(() => CustomReservationList, {
    description: `Get the custom reservation list configuration\nCapability required: \`custom_reservation_list\``,
    nullable: true,
    deprecationReason: 'Use restaurant.customReservationList',
  })
```

I think something changed on the `printSchema` utility function.

**Comment by MukundaKatta** [comment]
Yeah this changed in v16 when description handling switched to using block-string serialization only when the content naturally fits. The workaround is to avoid the leading whitespace in your template literal — graphql-js trims consistent leading whitespace from each line, so if you indent the whole block the trim logic can't kick in. A tagged dedent helper (or `dedent` from npm) around your description string gives you back the v15 behaviour cleanly.

**Comment by yudin-s** [comment]
This usually comes from the distinction between a GraphQL description block and the string value stored in the AST.

If you write:

```graphql
\"\"\"
Line one
Line two
\"\"\"
type Example {
  field: String
}
```

the parser stores a normalized string description. When you print the schema again, the printer decides how to render that string back into SDL. It is not a byte-for-byte preservation of the original source formatting.

If you need stable formatting, I would normalize descriptions before building/printing the schema:

```ts
import { stripIndent } from 'common-tags'

description: stripIndent`
  Line one
  Line two
`
```

The key is to avoid accidental leading indentation in the actual string value. Once that indentation is part of the description string, the printer has to preserve it as content.

So I would treat printed SDL as canonical output, not as a formatter that preserves the original description layout.
